flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/plan/rules/logical/PushFilterIntoLegacyTableSourceScanRuleTest.scala

```diff
@@ -196,4 +196,17 @@ class PushFilterIntoLegacyTableSourceScanRuleTest extends TableTestBase {
         |  TIMESTAMPADD(YEAR, 2, b) >= a
         |""".stripMargin)
   }
+
+  @Test
+  def testWithNullLiteral(): Unit = {
```

Comment:
I would prefer to add this test to another test class since we don't know when the legacy table source will be dropped.